src/hooks/xcm/useXcmBridge.ts

```diff
@@ -227,68 +261,80 @@ export function useXcmBridge(selectedToken: Ref<ChainAsset>) {
 
     // check if recipient account has non-zero native asset. (it cannot be transferred to an account with 0 nonce)
     isDisabledBridge.value =
-      !amount.value ||
-      Number(amount.value) === 0 ||
-      balance.value.lten(0) ||
-      errMsg.value !== '' ||
-      !isFulfilledAddress;
+      !amount.value || Number(amount.value) === 0 || errMsg.value !== '' || !isFulfilledAddress;
   };
 
   const setIsNativeBridge = (isNative: boolean): void => {
     resetStates();
     isNativeBridge.value = isNative;
   };
 
-  const connectRelaychain = async (): Promise<void> => {
-    let endpoint;
-    if (currentNetworkIdx.value === endpointKey.ASTAR) {
-      endpoint = xcmProviderEndpoints[xcmEndpointKey.POLKADOT].endpoint;
-    } else {
-      endpoint = xcmProviderEndpoints[xcmEndpointKey.KUSAMA].endpoint;
-    }
+  const connectOriginChain = async (): Promise<void> => {
+    const endpoint = xcmProviderEndpoints.find((it) => {
+      if (isAstarNativeTransfer.value) {
+        const symbol = String(selectedToken.value.metadata.symbol) as 'ASTR' | 'SDN' | 'SBY';
+        const defaultParachain = xcmAstarNativeToken[symbol].parachains![0];
+        return it.networkAlias.toLowerCase() === defaultParachain.toLowerCase();
+      } else {
+        return it.networkAlias.toLowerCase() === selectedToken.value.originChain.toLowerCase();
+      }
+    })?.endpoint as string;
 
     try {
-      relayChainApi = new RelaychainApi(endpoint);
-      await relayChainApi.start();
+      if (
+        originChain.value.name === 'Karura' ||
```

Comment:
of course, we need refactor this later. but just noted, `AcalaApi` should be separate with `useXcmBridge` hook because of high coupling with the general hook I think

Reply:
maybe refactor using with factory method pattern ? https://www.dofactory.com/javascript/design-patterns/factory-method but I believe it's scoped in @bobo-k2 's refactor task